services/director-v2/src/simcore_service_director_v2/modules/db/repositories/services_environments.py

```diff
@@ -0,0 +1,46 @@
+from models_library.services import ServiceKey
+from models_library.users import UserID
+from pydantic import EmailStr, parse_obj_as
+from simcore_postgres_database.utils_services_environments import (
+    VENDOR_SECRET_PREFIX,
+    VendorSecret,
+    get_vendor_secrets,
+)
+from simcore_postgres_database.utils_users import UsersRepo
+
+from ._base import BaseRepository
+
+
+class ServicesEnvironmentsRepository(BaseRepository):
+    """
+    Access to Vendor settings for a service
+    """
+
+    async def get_vendor_secrets(
+        self, service_key: ServiceKey
+    ) -> dict[str, VendorSecret]:
+        """Fetches vendor secrets for a service using normalized names"""
+        async with self.db_engine.acquire() as conn:
+            vendor_secrets: dict[str, VendorSecret] = await get_vendor_secrets(
+                conn, vendor_service_key=service_key, normalize_names=True
+            )
+
+            # NOTE: normalize_names = True
+            assert all(  # nosec
+                self.is_vendor_secret_identifier(key) for key in vendor_secrets
+            )  # nosec
```

Comment:
I would also transform this in a check rather than have an assert

Reply:
This is guaranteed by the utils function